src/Features/Core/Portable/EmbeddedLanguages/Json/JsonOptions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Features.EmbeddedLanguages.Json;
+
+[Flags]
+internal enum JsonOptions
+{
+    /// <summary>
+    /// Parse using loose rules.  This is very relaxed and allows lots of features not allowed by standard IETF 8259
+    /// (like comments, and trailing commas).  This also allows all the additional constructs added by Json.net, like
+    /// constructors and string that use single-quotes instead of double-quotes.
+    /// </summary> 
+    Loose = 0,
```

Comment:
Should this be structured that "Loose" is equal to Comments | TrailingCommas and strict = 0? Otherwise, it seems odd that some bits mean "be more strict" and some mean "be more loose"? I guess though this has to work this way because this is how the comment detector is looking for these particular names?

(in general this generic parser seems a bit janky here, since I guess a user could write loose,strict and we'd just go with that?)

Reply:
so... there's two issues here.  First: i want `// lang=json` to be as permissive as possible.  Which means the 0 state should be that (though i suppose we could also have a way to pass the 'default' that a detector should pick if no other information is provided).

Second, there are actually two separate groups.  The first choice is: "are we going json.net parsing, or .net parsing?"   Second, if we're doing .net parsing, is it in strict mode (teh default), or does it allow these two variants.

Note that Loose is still a *superset* of that, allowing a broad set of *other* craziness that is not controllable. 

I don't have a great way to model this IMO, but i'm open to ideas.